lib/kcerts/ssh.go

```diff
@@ -190,22 +249,48 @@ func FindSSHAgent(store cache.Store, logger logger.Logger) (*SSHAgent, error) {
 
 // FindSSHAgentFromEnv
 func FindSSHAgentFromEnv() *SSHAgent {
+	// If the SSH agent was started locally, both SSH_AGENT_SOCK and
+	// SSH_AGENT_PID will be set.  However, when using ssh-agent forwarding over
+	// an SSH or CRD session, only SSH_AUTH_SOCK will be set.
 	envSSHSock := os.Getenv("SSH_AUTH_SOCK")
 	envSSHPID := os.Getenv("SSH_AGENT_PID")
-	if envSSHSock == "" || envSSHPID == "" {
+	if envSSHSock == "" {
 		return nil
 	}
-	pid, err := strconv.Atoi(envSSHPID)
-	if err != nil {
-		return nil
+	pid := 0
+	if envSSHPID != "" {
+		var err error
+		pid, err = strconv.Atoi(envSSHPID)
+		if err != nil {
+			return nil
+		}
 	}
 	return &SSHAgent{PID: pid, Socket: envSSHSock, Close: func() {}}
 }
 
 // CreateNewSSHAgent creates a new ssh agent. Its env variables have not been added to the shell. It does not maintain
 // its own connection.
 func CreateNewSSHAgent() (*SSHAgent, error) {
-	cmd := exec.Command("ssh-agent", "-s")
+	user, err := UserCurrent()
+	if err != nil {
+		return nil, err
+	}
+	path := fmt.Sprintf("/tmp/ssh-%s", user.Username)
```

Comment:
make this a method on the agent? the filepath/mkdir/etc as well, so it's re-used.

Reply:
Will do.